### PR TITLE
Implemented exists related methods in reactive aerospike repository 

### PR DESCRIPTION
**Pull request by igor-ermolenko**
@Aloren could you please review this PR? Thank you!

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java [hunk 1]
```diff
@@ -185,8 +195,9 @@ public String getNamespace() {
         Assert.notNull(query, "Query must not be null!");
         Assert.notNull(type, "Type must not be null!");
 
-        Qualifier qualifier = query.getCriteria().getCriteriaObject();
-        return findAllRecordsUsingQuery(type, null, qualifier);
+        return isNull(query.getCriteria()) ?
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: Please check `findAllUsingQuery` method. It also  calls `query.getCriteria().getCriteriaObject()`, shouldn't we add the same check there?

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java [hunk 2]
```diff
@@ -185,8 +195,9 @@ public String getNamespace() {
...
+                findAllRecordsUsingQuery(type, (Filter) null) :
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: This PR is adding exists method implementation. Not sure how this fix is related to exists related methods? Could you please also point to the test that covers this fix.

File: src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java [hunk 3]
```diff
@@ -68,17 +70,21 @@
...
-        throw new UnsupportedOperationException("Method not implemented yet.");
+        Assert.notNull(publisher, "The given id must not be null!");
+
+        return Mono.from(publisher).flatMap(id -> operations.exists(id, entityInformation.getJavaType()));
     }
 
     @Override
     public Mono<Long> count() {
-        throw new UnsupportedOperationException("Method not implemented yet.");
+        return operations.count(new Query((Sort) null), entityInformation.getJavaType());
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java]: In non reactive repository `SimpleAerospikeRepository`:
```
	@Override
	public long count() {
		return operations.count(entityInformation.getJavaType());
	}
```
Why reactive implementation creates `new Query((Sort) null)`? Can we make reactive and non reactive implementations similar?

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java (hunk 1 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: Good point! Thank you! 
Verification added into findAllUsingQuery() method/

File: src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java (hunk 3 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/repository/support/SimpleReactiveAerospikeRepository.java]: Done.
Now reactive and non reactive implementations are the same.

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java (hunk 2 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: This fix needed for count() method. I did not make the PR separately for one method.
`    public Mono<Long> count() {
        return operations.count(entityInformation.getJavaType());
    }
`
Also I've added test for this case:
`
ReactiveAerospikeTemplateCountRelatedTests.countWithoutQuery_ShouldWorksProperly()
ReactiveAerospikeTemplateCountRelatedTests.count_ShouldCalculateOnlyItemsWithSpecifiedTypes()
`

**Comment by igor-ermolenko** [comment]
@Aloren, I've removed count() method implementation from reactive repository.

**Pull request merged**